src/main/java/org/terasology/climateConditions/BodyTemperatureSystem.java

```diff
@@ -0,0 +1,107 @@
+        for (EntityRef entity : entityManager.getEntitiesWith(WorldComponent.class)) {
+            if (!processedOnce) {
+                delayManager.addPeriodicAction(entity, BODY_TEMPERATURE_UPDATE_ACTION_ID, 0, CHECK_INTERVAL);
+                processedOnce = true;
+            } else {
+                logger.warn("More than one entity with WorldComponent found");
+            }
+        }
+        // The world is the entity here which shall have a Body Temperature Update Periodic Action.
+    }
+
+    @ReceiveEvent
+    public void onTemperatureUpdate(PeriodicActionTriggeredEvent event, EntityRef world) {
+        if (event.getActionId().equals(BODY_TEMPERATURE_UPDATE_ACTION_ID)) {
+            for (EntityRef entity : entityManager.getEntitiesWith(AliveCharacterComponent.class)) {
+                LocationComponent location = entity.getComponent(LocationComponent.class);
+                if (!entity.hasComponent(BodyTemperatureComponent.class)) {
+                    BodyTemperatureComponent btc = new BodyTemperatureComponent();
+                    entity.addComponent(btc);
+                }
+                BodyTemperatureComponent btc = entity.getComponent(BodyTemperatureComponent.class);
+                float envTemperature = climateConditionsSystem.getTemperature(location.getLocalPosition().getX(),
+                        location.getLocalPosition().getY(), location.getLocalPosition().getZ());
+                float envHumidity = climateConditionsSystem.getHumidity(location.getLocalPosition().getX(),
+                        location.getLocalPosition().getY(), location.getLocalPosition().getZ());
+                btc.bodyTemperature =
+                        btc.lastBodyTemperature + ((((envTemperature - (envHumidity / 10)) - btc.lastBodyTemperature) / 100000) * CHECK_INTERVAL);
+                btc.lastBodyTemperature = btc.bodyTemperature;
+                entity.saveComponent(btc);
+            }
+        }
+    }
+
+    @ReceiveEvent
+    public void onTemperatureDisplay(PeriodicActionTriggeredEvent event, EntityRef player, LocationComponent location
+            , BodyTemperatureComponent btc) {
+        // temporary: for help in development
+        if (event.getActionId().equals(BODY_TEMPERATURE_DISPLAY_ACTION_ID)) {
+            float envTemperature = climateConditionsSystem.getTemperature(location.getLocalPosition().getX(),
+                    location.getLocalPosition().getY(), location.getLocalPosition().getZ());
+            player.getOwner().send(new ChatMessageEvent("Body Temperature: " + btc.bodyTemperature, player.getOwner()));
+            player.getOwner().send(new ChatMessageEvent("Env Temperature: " + envTemperature, player.getOwner()));
+        }
+
+    }
+
+    @ReceiveEvent
+    public void onPlayerSpawn(OnPlayerSpawnedEvent event, EntityRef player) {
+        delayManager.addPeriodicAction(player, BODY_TEMPERATURE_DISPLAY_ACTION_ID, 0, DISPLAY_INTERVAL);
+        // temporary: for help in development
```

Comment:
why though? you can simply display the temperatures in `onTemperatureUpdate` before/after you update the body temperature, right? :thinking:

Reply:
yes, but every second was a bit too much...

Reply:
oh I see, okay, makes sense - we just need to remember getting rid of it again before merging this :wink: